CRM/Financial/Form/FrontEndPaymentFormTrait.php

```diff
@@ -50,12 +50,10 @@ protected function alterLineItemsForTemplate($tplLineItems) {
     $getTaxDetails = FALSE;
     foreach ($tplLineItems as $key => $value) {
       foreach ($value as $k => $v) {
-        if (isset($v['tax_rate'])) {
-          if ($v['tax_rate'] != '') {
-            $getTaxDetails = TRUE;
-            // Cast to float to display without trailing zero decimals
-            $tplLineItems[$key][$k]['tax_rate'] = (float) $v['tax_rate'];
-          }
+        if (isset($v['tax_rate']) && $v['tax_rate'] != '') {
```

Comment:
This is fine.  I fell into a trap which @seamuslee001 pointed out that you could have a 0% tax rate specifically defined which is why we don't want to collapse to empty.  Maybe worth a comment (for me next time I'm here) `// Don't use empty as a 0% tax rate may be ok.`